### ISSUE TITLE
skip tls?

### ISSUE THREAD

**Issue by bcg62** [poseidon/terraform-render-bootkube#37]
Looking for a way to have the module skip tls if keys/certs already exist. 

This is useful in situations where you need to re-generate bootkube for kubernetes upgrades.

It's also an issue if you wish to expand an existing cluster adding more nodes.

I know you can simply target bootkube in terraform but you still have to manually manage your secrets as they blown away on every regeneration.
 
terraform apply -target=module.cluster.module.bootkube

**Comment by dghubble** [comment]
I don't expect the Terraform "local_file" provider to override an existing file, if you've really gone to the trouble of keeping those files present in the target assets directory, but removed other files you're trying to have Terraform regenerate. If so, that'd be interesting to raise with the provider.

Typhoon now documents https://typhoon.psdn.io/topics/maintenance/#upgrades and dictates blue-green replacement as the cluster upgrade strategy. Rather than messing with partial or in-place edits of live clusters.

Adding new worker nodes is already supported and shouldn't be a factor here.

**Comment by bcg62** [comment]
Yes, it appears thats whats happening. Any time i run terraform apply i seem to be getting all new certs/keys.

**Comment by dghubble** [comment]
Has the version of your tls provider plugin changed? I may have seen this recently, something changes such that Terraform believes it needs to regenerate the certificate and local_file faithfully lays down the new contents since its agnostic. That seems like a more likely explanation than my original concern.

Some Terraform resources handle this by having a field for "ignore_changes". For example, the AWS instance resource has it to ignore ami changes that would reprovision a stateful node. Just some ideas / theories.

**Comment by bcg62** [comment]
It's possible, there's been circumstances i've flushed `.terraform` and reinitialized, I can't be positive this is the exact cause. Id hope even if this was the case I could preserve what I have in place.
